### PR TITLE
Implement `_grunt_chatter_mp.gnut`

### PR DESCRIPTION
**Pull request by Zanieon**
Code is adapted from `_grunt_chatter.gnut` which is used in the campaign.

File implementation is a vanilla behavior restoration of Grunts being able to chatter about when other grunts nearby are killed, or when an enemy Titan is killed.

### CODE REVIEW

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks pretty good, some examples of testing would be nice if possible 

(oh also accidental whitespace change in another file)

**Comment by Zanieon** [comment]
For testing with this file is super easy, just go into random Attrition and kill an enemy Grunt, some other grunt nearby will probably speak about the one you just killed, leave only one of their squad alive, and he will probably yell that he's alone and need backup.

File: Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut [hunk 1]
```diff
@@ -15,4 +23,177 @@ void function PlayGruntChatterMPLine( entity grunt, string conversationType )
...
+	// if another grunt from another squad is nearby, don't chatter about being alone
+	array<entity> nearbyGrunts = GetNearbyFriendlyGrunts( lastSquadMember.GetOrigin(), lastSquadMember.GetTeam(), CHATTER_SQUAD_DEPLETED_FRIENDLY_NEARBY_DIST )
+	nearbyGrunts.fastremovebyvalue( lastSquadMember )
+	if ( nearbyGrunts.len() )
+		return
+
+	PlayGruntChatterMPLine( lastSquadMember, "bc_squaddeplete" )
+}
+
+void function GruntChatter_TryEnemyDown( entity deadGuy )
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut]:
```suggestion
void function GruntChatter_TryEnemyTitanDown( entity deadGuy )
```
Just to make the naming a bit clearer

File: Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut [hunk 2]
```diff
@@ -15,4 +37,181 @@ void function PlayGruntChatterMPLine( entity grunt, string conversationType )
...
+
+	if ( grunt.ContextAction_IsMeleeExecution() )
+		return false
+
+	string squadname = expect string( grunt.kv.squadname )
+	// we only care about this because the grunt conversation system wants it
+	if ( squadname == "" )
+		return false
+
+	return true
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut:204]:
```suggestion
	return squadname == ""
```

File: Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut [hunk 3]
```diff
@@ -15,4 +37,181 @@ void function PlayGruntChatterMPLine( entity grunt, string conversationType )
...
+bool function GruntChatter_IsGruntTypeEligibleForChatter( entity grunt )
+{
+	if ( !IsGrunt( grunt ) )
+		return false
+
+	// mechanical grunts don't chatter
+	if ( grunt.IsMechanical() )
+		return false
+
+	return true
```
Comment (uniboi) [inline Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut:216]:
```suggestion
	return !grunt.isMechanical()
```

**Review by uniboi — APPROVED** [review]
Some small improvements, looks good otherwise

File: Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut (hunk 2 above)
Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut:204]: surely that should be !=

Comment (uniboi) [reply Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut:204]: womp womp

Comment (Zanieon) [reply Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut:204]: Bool function, it wont return a string

Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut:204]: > Bool function, it wont return a string

yeah thats the point of the `!= ""`

File: Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut (hunk 3 above)
Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/conversation/_grunt_chatter_mp.gnut]: There was a typo in that, caused compile check fail, `return !grunt.IsMechanical()` should work

**Pull request merged**